### PR TITLE
Uyuni salt config modules

### PR DESCRIPTION
**Pull request by moio**
# Description

This documents a new set of Salt states we are adding, which allow users to configure a server with organizations, users and system groups.

This is part of the Hub initiative, although those modules can also be used for single servers (arguably with less advantages to WebUI/API/spacecmd configuration).

Note: this feature in technical preview. 

# Target branches

Which product version does this PR apply to?

- [ ] Manager-4.1
- [ ] Manager-4.2
- [x] Uyuni

# Links

Part of https://github.com/SUSE/spacewalk/issues/9876

Supersedes https://github.com/uyuni-project/uyuni-docs/pull/482

### CODE REVIEW

**Comment by moio** [comment]
@Loquacity: FMPOV this is ready for review now. In order to avoid duplication and given this is only a tech preview at this point, we opted to leave most material in just one place, in the package.

For your reference, material is here: https://github.com/uyuni-project/uyuni/tree/master/susemanager-utils/susemanager-sls/src/doc

We could discuss over the best choice here: in the package, in docs or both.

From a scheduling perspective, this is to be delivered in the next 4.1 maintenance update.

TIA

**Comment by Loquacity** [comment]
OK, here's what I did:

* Edited the content
* Renamed the file to remove Uyuni
* Added to the nav (with an Uyuni conditional, so it doesn't show up in SUMA. Not sure if this was intended, if it's going in to SUMA 4.1.2 though?)
* Added the changelog

**Comment by Loquacity** [comment]
Also: do we need something in the LSD book to xref to this?

File: modules/salt/nav-salt-guide.adoc [hunk 1]
```diff
@@ -17,6 +17,9 @@ endif::[]
 ** xref:salt-file-locations.adoc[File Locations]
 ** xref:salt-gitfs.adoc[GitFS Fileserver Backend]
 ** xref:yomi.adoc[Install Using Yomi]
+ifeval::[{uyuni-content} == true]
```
Comment (moio) [inline modules/salt/nav-salt-guide.adoc:20]: This is going to go into 4.1.2 as technical preview, so this content is not Uyuni specific.

File: modules/salt/pages/config-modules.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
+[[config-modules]]
+= Configuration Modules
+
+[IMPORTANT]
+====
+This feature is a technology preview.
+====
+
+You can use Salt execution and state modules to configure your {productname} Server.
```
Comment (moio) [inline modules/salt/pages/config-modules.adoc:9]: Hmm, not sure if this is the best wording.

So these modules are called "Uyuni configuration modules" and they can target both Uyuni and SUSE Manager Servers.

As it is, this sentence seems to refer to generic Salt execution and state modules: of course one could use them to do whatever, including Server configuration, but here we are presenting some specific new modules that are tailored for that scope.

I'd still call them Uyuni configuration modules (just as the package name) and clarify they also apply to SUSE Manager contexts.

File: modules/salt/pages/config-modules.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+For more information about using Hub, see xref:large-deployments:multi-server.adoc[].
+
+You can use configuration modules to configure:
+
+* Organizations
+* Users
+* User permissions
+* System groups
+
+More configuration modules are planned to be released in the future.
```
Comment (moio) [inline modules/salt/pages/config-modules.adoc:22]: Technically speaking there is no plan set in stone yet, although I can see them coming. Not even sure if it's worth it to mention here...

File: modules/salt/pages/config-modules.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+* For more information about Salt execution modules and how to use them, see https://docs.saltstack.com/en/latest/topics/tutorials/modules.html.
+* For more information about Salt state modules, see https://docs.saltstack.com/en/latest/topics/tutorials/starting_states.html.
+
+
+
+== Install Configuration Modules
+
+The Salt modules are available in the [path]``uyuni-config-modules`` package.
```
Comment (moio) [inline modules/salt/pages/config-modules.adoc:31]: Again this sentence does not sound correct: [many](https://docs.saltstack.com/en/latest/ref/states/all/index.html) [many](https://docs.saltstack.com/en/latest/ref/modules/all/index.html) Salt modules exist and most of them are not part of that package. The sentence is supposed to refer only to the Uyuni configuration states.

**Comment by moio** [comment]
> Also: do we need something in the LSD book to xref to this?

It's not mandatory but it would help discoverability. This is not an LSD topic per se, but it is important when using multiple Servers.

File: modules/salt/pages/config-modules.adoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+You can use configuration modules to configure:
+
+* Organizations
+* Users
+* User permissions
+* System groups
+
+More configuration modules are planned to be released in the future.
+
+* For more information about Salt execution modules and how to use them, see https://docs.saltstack.com/en/latest/topics/tutorials/modules.html.
```
Comment (keichwa) [inline modules/salt/pages/config-modules.adoc:24]:
```suggestion
* For more information about Salt execution modules, see https://docs.saltstack.com/en/latest/topics/tutorials/modules.html.
```

File: modules/salt/pages/config-modules.adoc (hunk 4 above)
Comment (keichwa) [reply modules/salt/pages/config-modules.adoc:31]: I think we have a `[package]` tag.

**Review by keichwa — APPROVED** [review]

File: modules/salt/nav-salt-guide.adoc (hunk 1 above)
Comment (Loquacity) [reply modules/salt/nav-salt-guide.adoc:20]: OK, let's remove the ifeval, then. I was confused.

File: modules/salt/pages/config-modules.adoc (hunk 2 above)
Comment (Loquacity) [reply modules/salt/pages/config-modules.adoc:9]: OK, that's what had me confused! I've updated the wording now, so hopefully it's clearer.

File: modules/salt/pages/config-modules.adoc (hunk 3 above)
Comment (Loquacity) [reply modules/salt/pages/config-modules.adoc:22]: Removed.

**Comment by Loquacity** [comment]
image
image

**Review by jcayouette — APPROVED** [review]

**Review by keichwa — APPROVED** [review]

File: modules/salt/pages/config-modules.adoc [hunk 6]
```diff
@@ -0,0 +1,41 @@
+[[config-modules]]
+= Configuration Modules
+
+[IMPORTANT]
+====
+This feature is a technology preview.
+====
+
+Salt execution and state modules are used to configure your installation.
```
Comment (moio) [inline modules/salt/pages/config-modules.adoc:9]:
```suggestion
Execution and state modules are the main concepts in Salt to define, apply and orchestrate configuration of your devices.
```

File: .changelog [hunk 7]
```diff
@@ -36,3 +36,4 @@ terminology for versioning.
 - Fix Azure command in Install Guide (thanks Rahul-CTS)
 - Fix broken links in Auto-Install Proxy in Client Cfg (thanks @shirocco88)
 - Adds Ubuntu 20.04 supported features for Uyuni in Client Cfg
+- Adds Config Modules to the Salt Guide as tech preview
```
Comment (moio) [inline .changelog:39]:
```suggestion
- Adds Uyuni Config Modules to the Salt Guide as tech preview
```

File: modules/salt/pages/config-modules.adoc [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+{productname} provides a set of modules called {uyuni} configuration modules, that you can use to configure both {susemgr} and {uyuni} Servers.
+
+You can use the {uyuni} configuration modules directly or using SLS files.
+They are are especially useful if you have multiple {productname} Servers, for example in Hub installations, but they are also useful for smaller installations.
+
+For more information about using Hub, see xref:large-deployments:multi-server.adoc[].
+
+You can use configuration modules to configure:
```
Comment (moio) [inline modules/salt/pages/config-modules.adoc:17]:
```suggestion
You can use {uyuni} configuration modules to configure:
```

**Review by moio — COMMENTED** [review]
I am suggesting a couple of small things, but overall this is ready to be merged FMPOV.

Thanks!

File: modules/salt/pages/config-modules.adoc (hunk 6 above)
Comment (Loquacity) [reply modules/salt/pages/config-modules.adoc]: Slightly edited this to make it clearer, what do you think?

Comment (moio) [reply modules/salt/pages/config-modules.adoc]: :+1:

**Pull request merged**